### ISSUE TITLE
Feat(eos_designs): Option to set RCF (Route Control Function) for BGP neighbor/peer group, and route redistrubtion 

### ISSUE THREAD

**Issue by buddy-bunny** [aristanetworks/ansible-avd#2397]
### Enhancement summary

Feat(eos_designs and eos_cli_config_gen): Option to define RCF functions and set RCF (Route Control Function) in the following application points:
1.) for BGP neighbor/peer group: 
      direction: inbound and outbound
      address family to be supported (for both directions): 
           IPv4 unicast / IPv6 unicast 
           IPV4 labeled unicast / IPV6 labeled unicast
2.) for BGP route redistribution  
         connected protocol, static protocol, IS-IS protocol

### Which component of AVD is impacted

eos_designs, eos_cli_config_gen

### Use case example

This is to be used in a L3LS deployment. Currently, the route-map is supported in router BGP but not RCF. Request to have the option for RCF as well as EOS will have more RCF enhancements than route-map moving forward.

### Describe the solution you would like

In general sense, similar to options implemented for route-map. 

### Describe alternatives you have considered

none available

### Additional context

RCF TOI: https://www.arista.com/en/support/toi/eos-4-24-1f/14605-routing-control-functions
Found that RCF has limited support under ISIS in AVD:  https://github.com/aristanetworks/ansible-avd/issues/1809

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

**Comment by ClausHolbechArista** [comment]
@Vibhu-gslab please reevaluate what is needed here. I think we have implemented a lot of it in 5.0, but there might still be some areas left.

**Comment by ClausHolbechArista** [comment]
The issue here is also covering eos_designs. Since we have everything in eos_cli_config_gen now I will just change the title to eos_designs.

**Comment by Vibhu-gslab** [comment]
> @Vibhu-gslab please reevaluate what is needed here. I think we have implemented a lot of it in 5.0, but there might still be some areas left.

Few missing RCF implementations:
```
address-family ipv4 unicast
   aggregate-address 192.0.2.0/24 attribute rcf AGGREGATE_FUNCTION()

router bgp 64500
   vrf VRF1
      address-family ipv4
         route-target import vpn-ipv4 rcf IMPORT_POLICY() vpn-route filter-rcf FILTER_VPN_ROUTES()
      address-family ipv6
         route-target import vpn-ipv6 rcf IMPORT_POLICY() vpn-route filter-rcf FILTER_VPN_ROUTES()

router bgp 64500
   vrf VRF1
      address-family ipv4
         route-target import evpn rcf IMPORT_POLICY()
      address-family ipv6
         route-target import evpn rcf IMPORT_POLICY()

router bgp 64500
   address-family ipv4
      neighbor 192.0.2.1 default-originate always rcf DEFAULT_ORIGINATE_POLICY()
   address-family ipv6
      neighbor 192.0.2.1 default-originate always rcf DEFAULT_ORIGINATE_POLICY()

router bgp 64500
   vrf VRF1
      address-family ipv4
         default-route export vpn-ipv4 always rcf DEFAULT_EXPORT_POLICY()
      address-family ipv6
         default-route export vpn-ipv6 always rcf DEFAULT_EXPORT_POLICY()

vrf VRF1
      address-family ipv4
         default-route export evpn always rcf DEFAULT_EXPORT_POLICY()
      address-family ipv6
         default-route export evpn always rcf DEFAULT_EXPORT_POLICY()
address-family ipv4 multicast
   route input address-family ipv4 unicast rcf ROUTE_INPUT_FUNCTION()

router bgp 64500
   address-family rt-membership
      neighbor 192.0.2.1 rcf out OUTBOUND_POLICY()

router bgp 64500
   address-family flow-spec ipv4
      neighbor 192.0.2.1 rcf in INBOUND_POLICY()
   address-family flow-spec ipv6
      neighbor 192.0.2.1 rcf in INBOUND_POLICY()

router bgp 64500
   address-family flow-spec vpn-ipv4
      neighbor 192.0.2.1 rcf out OUTBOUND_POLICY()
   address-family flow-spec vpn-ipv6
      neighbor 192.0.2.1 rcf out OUTBOUND_POLICY()

router bgp 64500
   address-family vpls
      neighbor 192.0.2.1 rcf out OUTBOUND_POLICY()

dynamic prefix-list DYN1
   match rcf TRIGGERING_ROUTE_EXISTS()
```

Please let me know if we need to include any of these in eos_cli_config_gen.

**Comment by ClausHolbechArista** [comment]
I don't think we will know for sure until we implement it in eos_designs, so I think we should just wait for that first.

**Comment by gmuloc** [comment]
Multiple steps:
1/ Find places in `eos_designs` where we are exposing `route_map_in`/`route_map_out` field (primarily in `network_services`) and add `rcf_in` / `rcf_out` - @Vibhu-gslab 

2/ in a second time we will need to add the option AVD built-in route-maps with RCFs (more work to do and need to prepare the rcfs)

**Comment by MaheshGSLAB** [comment]
> Multiple steps: 1/ Find places in `eos_designs` where we are exposing `route_map_in`/`route_map_out` field (primarily in `network_services`) and add `rcf_in` / `rcf_out` - [@Vibhu-gslab](https://github.com/Vibhu-gslab)
Raised the PR and merged(https://github.com/aristanetworks/avd/pull/4804)
> 
> 2/ in a second time we will need to add the option AVD built-in route-maps with RCFs (more work to do and need to prepare the rcfs)

**Comment by gmuloc** [comment]
Issue on hold until we get better clarity for step 2/

**Comment by momorientes** [comment]
Any progress on this? We would also be curious to see support for RCF in AVD. We're doing a lot of eBGP peering with IXPs/Transits and currently have to rely on fully generated route_maps which results in a extremely bloated configuration.

**Comment by Nielszy** [comment]
I would also be very interested in this functionality for the same reasons as @momorientes.
